esi_leap/api/controllers/v1/contract.py

```diff
@@ -102,19 +114,41 @@ def post(self, new_contract):
         return Contract(**c.to_dict())
 
     @wsme_pecan.wsexpose(Contract, wtypes.text)
-    def delete(self, contract_uuid):
+    def delete(self, contract_id):
         request = pecan.request.context
         cdict = request.to_policy_values()
         policy.authorize('esi_leap:contract:delete', cdict, cdict)
 
-        c = contract.Contract.get(request, contract_uuid)
-        if c.project_id != request.project_id:
-            o = offer.Offer.get(request.project_id, c.offer_uuid)
-            if o.project_id != request.project_id:
+        c_id = contract.Contract.get(contract_id)
+
+        permitted = []
+        for c in c_id:
+            try:
+                ContractsController._contract_authorize_management(c, cdict)
+                permitted.append(c)
+            except oslo_policy.PolicyNotAuthorized:
+                continue
+
+            if len(permitted) > 1:
+                raise exception.ContractDuplicateName(name=contract_id)
+
+        if len(permitted) == 0:
+            raise exception.ContractNotFound(contract_id=contract_id)
+
```

Comment:
This code - 122 to 136 - could probably be in its own function. I might recommend putting this all in _contract_authorize_management and possibly renaming that function.